wire-ios-utilities/Source/Public/OptionalComparison.swift

```diff
@@ -18,17 +18,31 @@
 
 import Foundation
 
-public func < <T: Comparable>(lhs: T?, rhs: T?) -> Bool {
-    switch (lhs, rhs) {
-    case let (l?, r?): return l < r
-    case (nil, _?): return true
-    default: return false
-    }
-}
+public enum OptionalComparison {
+    /// Compares two optional values in ascending order.
+    ///
+    /// - Parameters:
+    ///   - lhs: An optional comparable value on the left hand side.
+    ///   - rhs: An optional comparable value on  the right hand side.
+    ///
+    /// - Returns:
+    ///  `true` if left hand side is smaller than right hand side.
+    /// `nil` is  considered as smaller than any value.
+    /// If both paratemeters are `nil`, the method returns `false`.
+    public static func prependingNilAscending<T: Comparable>(lhs: T?, rhs: T?) -> Bool {
```

Comment:
The naming can be discussed.. it was not easy for me to express what it does in a comprehensive way. Thanks also to @caldrian for the proposal.

Reply:
I actually thought about an extension to `Collection`, something like this:

```
extension Collection {

    func sortedAscendingPrependingNil<Wrapped>() -> [Element] where Element == Optional<Wrapped>, Wrapped: Comparable {
        sorted { lhs, rhs in

            if lhs == nil, rhs == nil {
                return false
            }

            guard let lhs else {
                return true
            }

            guard let rhs else {
                return false
            }

            return lhs < rhs
        }
    }
}
```

Reply:
Alternatively, to keep the syntax similar and let the user provide a closure:
```
extension Collection {

    func sortedPrependingNil<Wrapped>(
        by areInIncreasingOrder: (Wrapped, Wrapped) throws -> Bool
    ) rethrows -> [Element]
    where Element == Wrapped?, Wrapped: Comparable {
        try sorted { lhs, rhs in

            if lhs == nil, rhs == nil {
                return false
            }

            guard let lhs else {
                return true
            }

            guard let rhs else {
                return false
            }

            return try areInIncreasingOrder(lhs, rhs)
        }
    }
}
```

Reply:
I am not sure to grasp what we gain compared to the custom operator? clarity?

Reply:
@netbe the reason why this change is important is, that one can not find a general solution to the question "how to compare nil with a value".

But with the custom operator it implicitly "works" without any compiler warning, which can lead to interesting cases like here:

1. Does `epoch=nil` enter the if with `true` or `false`

<img width="844" alt="Screenshot 2024-05-14 at 16 41 29" src="https://github.com/wireapp/wire-ios/assets/7196958/4be38271-18d3-4e46-94e6-ddef170a821d">

2. What if on of the dates is nil? what if both values are `nil`?
<img width="852" alt="Screenshot 2024-05-14 at 16 46 39" src="https://github.com/wireapp/wire-ios/assets/7196958/b7b8a837-49e9-4233-90e3-050ee890c045">

Reply:
Nice, `sortedAscendingPrependingNil` is very clear about the handling of `nil`.